db/src/tx.rs

```diff
@@ -669,6 +671,7 @@ impl<'conn, 'a, W> Tx<'conn, 'a, W> where W: TransactWatcher {
                     }
 
                     self.watcher.datom(op, e, a, &v);
+                    affected_attrs.insert(a);
```

Comment:
The reason I added the `watcher` was to abstract away this growing set of "when this happens inside transact, do this" behaviors. Imagine that we want to add an observer that fires if a particular value is seen, a particular entity is touched, or any datom is retracted.

You can easily imagine the observer API you've built being expressed as a watcher that collects notes about `tx_id`s, later (when the tx is committed) using those notes to dispatch particular `TxReport`s to particular observers. Indeed, you can use an arbitrary datom-level predicate to decide whether a report is of interest.

You don't have to make this change now, but it would be good to leave a TODO here (and on `changeset`) to say that it's subject to change.